src/test/suite/fixtures/recorded/languages/java/takeTypeGust2.yml

```diff
@@ -0,0 +1,39 @@
+languageId: java
+command:
+  version: 1
+  spokenForm: take type gust
+  action: setSelection
+  targets:
+    - type: primitive
+      modifier: {type: containingScope, scopeType: type, includeSiblings: false}
+      mark: {type: decoratedSymbol, symbolColor: default, character: g}
+initialState:
+  documentContents: |-
+    public class MyClass {
+        private MyClass () {
+            Map<String, String> map = new HashMap<String, String>();
+            List<String> list = new ArrayList<String>();
+        }
+    }
+  selections:
+    - anchor: {line: 3, character: 32}
+      active: {line: 3, character: 49}
+  marks:
+    default.g:
+      start: {line: 3, character: 42}
+      end: {line: 3, character: 48}
+finalState:
+  documentContents: |-
+    public class MyClass {
+        private MyClass () {
+            Map<String, String> map = new HashMap<String, String>();
+            List<String> list = new ArrayList<String>();
```

Comment:
I don't think this behaviour is consistent with typescript. I believe in typescript it would select `List<String>` here. Can you double check that? I'd argue for consistency across languages to the extent that it makes sense

Are you arguing to have this behaviour in Java but the other in Typescript? Or to have this behaviour in both Java and Typescript?

In my mind the semantics of "type" are that it takes the type annotation of the current scope. I recognise that it is already a bit overloaded though because it can be used to refer to a type / interface declaration

Reply:
I would like to have this behavior in both java and typescript. I have never used a generic class in typescript yet, but doing a test I believe it's incorrect in typescript as it is implemented now.

Exactly. The current scope. If your selection is in the value/new expression there,s a type there as well. And I think it's very useful being able to take both types.

Reply:
I would lean towards today's typescript behaviour, but I'm getting the impression you feel more strongly than I do, so I'll yield 😊.  Mind either changing typescript or at least filing an issue to do so in the future?